File: LoochaMediaLibrary/src/com/realcloud/media/loader/MediaBaseLoader.java
package com.realcloud.media.loader;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.LoaderManager;
import android.support.v4.content.Loader;
import android.support.v4.content.LoochaAsyncTaskLoader;

/**
 * Created by zack on 2017/9/29.
 */

public abstract class MediaBaseLoader<E> extends LoochaAsyncTaskLoader<E> implements LoaderManager.LoaderCallbacks<E> {

    private Bundle bundleArgs;

    public Bundle getBundleArgs() {
        return bundleArgs;
    }

    public MediaBaseLoader(Context context) {
        super(context);
    }

    @Override
    protected DATA_SOURCE getDataSource() {
        return DATA_SOURCE.LOCAL;
    }

    @Override
    protected void onStartLoading() {
        forceLoad();
    }

    @Override
    protected void onStopLoading() {
        super.onStopLoading();
        cancelLoad();
    }

    @Override
    protected void onReset() {
        super.onReset();
        onStopLoading();
    }

    @Override
    public Loader<E> onCreateLoader(int i, Bundle bundle) {
        this.bundleArgs = bundle;
        return this;
    }

    @Override
    public void onLoaderReset(Loader<E> loader) {

    }

}

File: MediaLibrary/src/main/java/com/zack/media/loader/MediaBaseLoader.kt
package com.zack.media.loader

import android.content.Context
import android.os.Bundle
import androidx.loader.app.LoaderManager
import androidx.loader.content.AsyncTaskLoader
import androidx.loader.content.Loader

/**
 * @Author zack
 * @Date 2019/7/26
 * @Description 基础loader
 * @Version 1.0
 */
abstract class MediaBaseLoader<T>(context: Context) : AsyncTaskLoader<T>(context), LoaderManager.LoaderCallbacks<T>{

    private var bundleArgs: Bundle? = null

    fun getBundleArgs(): Bundle? {
        return bundleArgs
    }

    override fun onCreateLoader(id: Int, args: Bundle?): Loader<T> {
        this.bundleArgs = args
        return this
    }

    override fun onStartLoading() {
        super.onStartLoading()
        forceLoad()
    }

    override fun onStopLoading() {
        super.onStopLoading()
        cancelLoad()
    }

    override fun onReset() {
        super.onReset()
        onStopLoading()
    }

    override fun onLoaderReset(loader: Loader<T>) {

    }
}